crates/rpc/rpc/src/eth/filter.rs

```diff
@@ -197,44 +197,50 @@ where
     ///  - underlying database error
     ///  - amount of matches exceeds configured limit
     #[allow(dead_code)]
-    fn filter_logs(
-        &self,
-        filter: &Filter,
-        _from_block: u64,
-        _to_block: u64,
-    ) -> RpcResult<Vec<Log>> {
-        let logs = Vec::new();
-        let topics = if filter.has_topics() {
-            let params = FilteredParams::new(Some(filter.clone()));
-            Some(params.flat_topics)
-        } else {
-            None
-        };
+    fn filter_logs(&self, filter: &Filter, from_block: u64, to_block: u64) -> RpcResult<Vec<Log>> {
+        let mut logs = Vec::new();
+        let filter_params = FilteredParams::new(Some(filter.clone()));
+
+        let topics =
+            if filter.has_topics() { Some(filter_params.flat_topics.clone()) } else { None };
 
-        let _address_filter = FilteredParams::address_filter(&filter.address);
-        let _topics_filter = FilteredParams::topics_filter(&topics);
-
-        // TODO blocked by <https://github.com/paradigmxyz/reth/issues/1371>
-        // let block_number = from_block;
-        //
-        // while block_number <= to_block {
-        //     let _block = self
-        //         .client
-        //         .block_by_number(BlockNumberOrTag::Number(block_number.into()))
-        //         .to_rpc_result()?
-        //         .ok_or(EthApiError::UnknownBlockNumber)?;
-        //
-        //
-        //
-        //     if FilteredParams::matches_address(block.header.logs_bloom, &address_filter) &&
-        //         FilteredParams::matches_topics(block.header.logs_bloom, &topics_filter)
-        //     {
-        //         // TODO filter logs from transactions
-        //     }
-        // }
+        // derive bloom filters from filter input
+        let address_filter = FilteredParams::address_filter(&filter.address);
+        let topics_filter = FilteredParams::topics_filter(&topics);
+
+        for block_number in from_block..=to_block {
```

Comment:
just checked that the range is indeed inclusive
https://github.com/ethereum/go-ethereum/blob/f86f04864603d4d1398e134fae319531453b67e9/core/bloombits/matcher.go#L193